### PR TITLE
Gzip compression option

### LINKED ISSUE #278: Up is compressing an already compressed response
Related to https://github.com/apex/up/issues/256#issuecomment-323809861

By default the proxy has gzip compression enabled.  
If the server has already a gzip compression the proxy is going to compress a compressed response leading to content decoding errors.

### PR DESCRIPTION
**Pull request by franciscocpg**
Add gzip compression option. Closes #278.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -432,6 +432,18 @@ The following settings are available:
...
 ```
+- `gzipCompression` - By default the proxy has gzip compression enabled.  
+If your server has already a gzip compression the proxy is going to compress a
+compressed response leading to content decoding errors.  
+So you may disable this compression by setting this flag to `false`.  
+```json
+{
+  "proxy": {
+    "command": "node app.js",
+    "gzipCompression": false
```
Comment (hhsnopek) [inline docs/configuration.md]: what if `gzipCompression` was `compression` and consumed `string` or `[]string` that way later expand the proxy compression types?

**Comment by stephenmathieson** [comment]
Couldn't the proxy check for encoding headers coming from origin to see if the response is already gzipped? Then just nop rather than compressing anything

**Comment by tj** [comment]
Hmm I agree I think we could get away with checking Content-Encoding and bypass transparently without an option.

**Comment by franciscocpg** [comment]
Ok I'll try to implement a middleware for that

**Comment by franciscocpg** [comment]
To be honest, after a few tests I'm not sure this is really needed right now.
Closing it.

**Pull request closed without merging**